duckduckgo/iOS
# Remote Messaging

Task/Issue URL: https://app.asana.com/0/72649045549333/1201884213179453/f
Tech Design URL: https://app.asana.com/0/481882893211075/1202238116542877/f
CC:

**Description**:
This PR is for the implementation of Remote Messaging.



**Steps to test this PR**:
1. No message during onboarding
	• Fresh install
	• Open the app and go through the onboarding screens
	• Ensure you don't see any remote message on the home tab if there's a Dax dialog
	• after completing the onboarding, relaunch the app and ensure first remote message appears
	• ensure pixel `m_remote_message_shown` and `m_remote_message_shown_unique`
	• visit any site, and open a new tab
	• Ensure message appears on the new home tab

2. New message
	• Now click on the "X" to dismiss the remote message
	• ensure pixel `m_remote_message_dismissed` sent
	• Relaunch the app 
	• Ensure a new message appears
	• Click on "Get tips"
	• Ensure it performs a search of "covid 19"
	• Ensure `m_remote_message_primary_action_clicked` pixel sent
	• Open a new tab
	• Ensure remote message is gone
	
3. Message with favorites
	• Add 2 sites as favorites. Open new tab and ensure favorites are visible
	• Relaunch the app
	• Ensure new message "app update" appears
	• Ensure remote message and favorites are visible
	• Click on "Later"
	• Ensure `m_remote_message_secondary_action_clicked` pixel sent
	• Ensure remote message is gone

4. Run unit tests and ensure they are green

![image](https://user-images.githubusercontent.com/91189922/182309806-68a7882d-7a98-46e8-a1e6-62001ba2c987.png) ![image](https://user-images.githubusercontent.com/91189922/182309649-38bc8313-88f9-4a1f-b93e-9b9dc355ec97.png) ![image](https://user-images.githubusercontent.com/91189922/182309669-cc7ed8ee-f854-42d8-b2e0-5a3e17c7529d.png) ![image](https://user-images.githubusercontent.com/91189922/182309681-dc54767b-3a32-4d17-a727-3c4acc2ed0b2.png)




**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
